### PR TITLE
Implementation of AzureRM End point (service connection) 

### PR DESCRIPTION
**Pull request by mikaelkrief**
## All Submissions:
-------------------------------------
* [YES] Have you added an explanation of what your changes do and why you'd like us to include them?
* [YES] I have updated the documentation accordingly.
* [YES] I have added tests to cover my changes.
* [YES] All new and existing tests passed.
* [YES] My code follows the code style of this project.
* [YES] I ran lint checks locally prior to submission.
* [YES] Have you checked to ensure there aren't other open PRs for the same update/change?

## What is the current behavior?
-------------------------------------

Issue Number: #161 

No Azure RM service connection

## What is the new behavior?
-------------------------------------

- Possibility to manage Azure Service Connection

## Does this introduce a breaking change?
-------------------------------------
- [YES]

## Any relevant logs, error output, etc?
-------------------------------------
(If it’s long, please paste to https://ghostbin.com/ and insert the link here.)

## Other information
-------------------------------------

### CODE REVIEW

**Comment by mikaelkrief** [comment]
@nmiodice : If you can merge the PR #211 before mine, because it's include the import methods for service endpoints.

File: azuredevops/resource_serviceendpoint_azurerm_test.go [hunk 1]
```diff
@@ -0,0 +1,274 @@
...
+var dhTestServiceEndpointAzureRMID = uuid.New()
+var dhRandomServiceEndpointAzureRMProjectID = uuid.New().String()
+var dhTestServiceEndpointAzureRMProjectID = &dhRandomServiceEndpointAzureRMProjectID
+
+var dhTestServiceEndpointAzureRM = serviceendpoint.ServiceEndpoint{ //todo change
+	Authorization: &serviceendpoint.EndpointAuthorization{
+		Parameters: &map[string]string{
+			"authenticationType":  "spnKey",
+			"scope":               "/subscriptions/fa8e7d5e-84f9-4477-904f-852054f85586",
+			"serviceprincipalid":  "e31eaaac-47da-4156-b433-9b0538c94b7e",
```
Comment (nmiodice) [inline azuredevops/resource_serviceendpoint_azurerm_test.go]: Are ` serviceprincipalid`, `serviceprincipalkey`, `tenantid`, `scope` and `SubscriptionId` real values? If they are, can we remove them? And if they are not, can you make them obviously fake?

File: azuredevops/utils/testhelper/hcl.go [hunk 2]
```diff
@@ -81,6 +81,25 @@ resource "azuredevops_serviceendpoint_dockerhub" "serviceendpoint" {
 	return fmt.Sprintf("%s\n%s", projectResource, serviceEndpointResource)
 }
 
+// TestAccServiceEndpointAzureRMResource HCL describing an AzDO service endpoint
+func TestAccServiceEndpointAzureRMResource(projectName string, serviceEndpointName string) string {
+	serviceEndpointResource := fmt.Sprintf(`
+resource "azuredevops_serviceendpoint_azurerm" "serviceendpointrm" {
+	project_id             = azuredevops_project.project.id
+	service_endpoint_name  = "%s"
+	azurerm_spn_clientid 	="e318e66b-ec4b-4dff-9124-41129b9d7150"
```
Comment (nmiodice) [inline azuredevops/utils/testhelper/hcl.go]: Can we pull these from the ENV instead? I am happy to add them to the AzDO environment we run internally if you can provide me a description of everything you need to be there.

File: azuredevops/resource_serviceendpoint_azurerm_test.go (hunk 1 above)
Comment (mikaelkrief) [reply azuredevops/resource_serviceendpoint_azurerm_test.go]: No it's fake values

File: azuredevops/resource_serviceendpoint_azurerm_test.go [hunk 3]
```diff
@@ -0,0 +1,274 @@
...
+	"github.com/google/uuid"
+
+	"github.com/microsoft/azure-devops-go-api/azuredevops/serviceendpoint"
+)
+
+var dhTestServiceEndpointAzureRMID = uuid.New()
+var dhRandomServiceEndpointAzureRMProjectID = uuid.New().String()
+var dhTestServiceEndpointAzureRMProjectID = &dhRandomServiceEndpointAzureRMProjectID
+
+var dhTestServiceEndpointAzureRM = serviceendpoint.ServiceEndpoint{ //todo change
```
[the remark this answers is not in the document]
Comment (mikaelkrief) [reply azuredevops/resource_serviceendpoint_azurerm_test.go]: No, it's old trace from copy/paste. it's removed.
+ I renamed variables

**Comment by camiloborges** [comment]
Hello guys.
I am keen to help getting this one across the line as I am building a demo that would benefit immensely from having this bit completed. Is there anyway I can help you guys out?
cheers

**Comment by mikaelkrief** [comment]
> Hello guys.
> I am keen to help getting this one across the line as I am building a demo that would benefit immensely from having this bit completed. Is there anyway I can help you guys out?
> cheers


@nmiodice can we remove the GitHub Action, for merge this PR ?

**Comment by nmiodice** [comment]
@mikaelkrief @camiloborges  So sorry for the delay. I've been quite busy finalizing the plans for my wedding, and OOTO for the past week to actually have my wedding. So sorry to leave this sitting.

It looks like after the merge of #211 there are a few conflicts. Please fix and then I will merge.

PS:
Thank you for your great work. Despite my lack of ability to devote time and effort over the past few weeks to this project, please be assured that your contributions are greatly appreciated. I'm back now, and should be able to keep more of an eye on things.

**Comment by nmiodice** [comment]
@mikaelkrief any update on resolving the conflicts?

**Comment by tmeckel** [comment]
@nmiodice Hi Nick! When can we expect this PR being merged? We also have a need for this feature in a project! Copy @tiwood for reference.

**Comment by nmiodice** [comment]
I've gone ahead and contributed back to the branch that @mikaelkrief is working on to fix the linting errors. Re-running build now.

**Comment by nmiodice** [comment]
Looks like there were a few other out of date things on this branch. Going through to try to fix today.

**Comment by mikaelkrief** [comment]
> Looks like there were a few other out of date things on this branch. Going through to try to fix today.

Thanks @nmiodice , sorry I am very busy on new job projects

**Comment by nmiodice** [comment]
@mikaelkrief it happens! no worries

**Comment by tmeckel** [comment]
@nmiodice Nick if you are able to merge this PR today this would be awesome, because me and @tiwood are really waiting for this feature!

**Comment by nmiodice** [comment]
@tmeckel Ask and you shall receive :)

**Comment by tmeckel** [comment]
Okay ... well then I ask another favor: could you check on the PR #129 as well 😀

**Pull request merged**